File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/video/RecordingStatsImpl.kt
package dev.zeekr.camerax_android.video

import androidx.camera.video.AudioStats
import androidx.camera.video.RecordingStats
import dev.zeekr.camerax_android.CameraXApiPigeonProxyApiRegistrar
import dev.zeekr.camerax_android.PigeonApiRecordingStatsApi

class RecordingStatsImpl(registrar: CameraXApiPigeonProxyApiRegistrar) : PigeonApiRecordingStatsApi(registrar) {
    override fun audioStats(pigeon_instance: RecordingStats): AudioStats {
        return pigeon_instance.audioStats
    }

    override fun numBytesRecorded(pigeon_instance: RecordingStats): Long {
        return pigeon_instance.numBytesRecorded
    }

    override fun recordedDurationNanos(pigeon_instance: RecordingStats): Long {
        return pigeon_instance.recordedDurationNanos
    }
}

File: camerax_android/lib/src/video/recording_stats_impl.dart
import 'package:camerax_android/src/api.dart';
import 'package:camerax_android/src/video.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class RecordingStatsImpl implements RecordingStats {
  final RecordingStatsProxyApi api;

  RecordingStatsImpl.internal(this.api);

  @override
  AudioStats get audioStats => api.audioStats.impl;
  @override
  int get numBytesRecorded => api.numBytesRecorded;
  @override
  Duration get recordedDuration =>
      Duration(microseconds: api.recordedDurationNanos ~/ 1000);
}

extension RecordingStatsProxyApiX on RecordingStatsProxyApi {
  RecordingStats get impl => RecordingStatsImpl.internal(this);
}
